### PR TITLE
codelimit 0.6.0 (new formula)

### PR DESCRIPTION
**Pull request by robvanderleek**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/c/codelimit.rb [hunk 1]
```diff
@@ -0,0 +1,185 @@
+class Codelimit < Formula
+  include Language::Python::Virtualenv
+
+  desc "Your Refactoring Alarm"
+  homepage "https://github.com/getcodelimit/codelimit"
+  url "https://files.pythonhosted.org/packages/71/db/872656169f3a75e923faf89c4ef57ccf7691722422e6d708f6ced6edbde2/codelimit-0.6.0.tar.gz"
+  sha256 "2791dded3c894a44525bcb9d354088626d1627272137b8273513bfb704efbfea"
+
```
Comment (p-linnane) [inline Formula/c/codelimit.rb:8]:
```suggestion
license "ISC"

```

File: Formula/c/codelimit.rb [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  bottle do
+    root_url "https://github.com/getcodelimit/homebrew-codelimit/releases/download/codelimit-0.6.0"
+    sha256 cellar: :any_skip_relocation, ventura:      "c7cfc66afff1c3aa8c944f0bd574563a5c328bf371384fb2f0659d68dce661c4"
+    sha256 cellar: :any_skip_relocation, x86_64_linux: "adf7735830c54a057bd4f0b6db0d9f02b99eb6ec928a9efb86cb8805e0ed4920"
+  end
+
```
Comment (p-linnane) [inline Formula/c/codelimit.rb:14]:
```suggestion

```

Comment (p-linnane) [reply Formula/c/codelimit.rb:14]: The bottle block will be generated by our workflows if this PR is accepted.

File: Formula/c/codelimit.rb [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  resource "attrs" do
+    url "https://files.pythonhosted.org/packages/97/90/81f95d5f705be17872843536b1868f351805acf6971251ff07c1b8334dbb/attrs-23.1.0.tar.gz"
+    sha256 "6279836d581513a26f1bf235f9acd333bc9115683f14f7e8fae46c98fc50e015"
+  end
+
+  resource "certifi" do
+    url "https://files.pythonhosted.org/packages/98/98/c2ff18671db109c9f10ed27f5ef610ae05b73bd876664139cf95bd1429aa/certifi-2023.7.22.tar.gz"
+    sha256 "539cc1d13202e33ca466e88b2807e29f4c13049d6d87031a3c110744495cb082"
+  end
+
```
Comment (p-linnane) [inline Formula/c/codelimit.rb:41]:
```suggestion

```

File: Formula/c/codelimit.rb [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+  resource "plotext" do
+    url "https://files.pythonhosted.org/packages/27/d7/58f5ec766e41f8338f04ec47dbd3465db04fbe2a6107bca5f0670ced253a/plotext-5.2.8.tar.gz"
+    sha256 "319a287baabeb8576a711995f973a2eba631c887aa6b0f33ab016f12c50ffebe"
+  end
+
+  resource "Pygments" do
+    url "https://files.pythonhosted.org/packages/d6/f7/4d461ddf9c2bcd6a4d7b2b139267ca32a69439387cc1f02a924ff8883825/Pygments-2.16.1.tar.gz"
+    sha256 "1daff0494820c69bc8941e407aa20f577374ee88364ee10a98fdbe0aece96e29"
+  end
+
```
Comment (p-linnane) [inline Formula/c/codelimit.rb:116]:
```suggestion

```

File: Formula/c/codelimit.rb [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+  resource "rich" do
+    url "https://files.pythonhosted.org/packages/ad/1a/94fe086875350afbd61795c3805e38ef085af466a695db605bcdd34b4c9c/rich-13.5.2.tar.gz"
+    sha256 "fb9d6c0a0f643c99eed3875b5377a184132ba9be4d61516a55273d3554d75a39"
+  end
+
+  resource "six" do
+    url "https://files.pythonhosted.org/packages/71/39/171f1c67cd00715f190ba0b100d606d440a28c93c7714febeca8b79af85e/six-1.16.0.tar.gz"
+    sha256 "1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926"
+  end
+
```
Comment (p-linnane) [inline Formula/c/codelimit.rb:131]:
```suggestion

```

File: Formula/c/codelimit.rb [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+  resource "typer" do
+    url "https://files.pythonhosted.org/packages/5b/49/39f10d0f75886439ab3dac889f14f8ad511982a754e382c9b6ca895b29e9/typer-0.9.0.tar.gz"
+    sha256 "50922fd79aea2f4751a8e0408ff10d2662bd0c8bbfa84755a699f3bada2978b2"
+  end
+
+  resource "typing-extensions" do
+    url "https://files.pythonhosted.org/packages/3c/8b/0111dd7d6c1478bf83baa1cab85c686426c7a6274119aceb2bd9d35395ad/typing_extensions-4.7.1.tar.gz"
+    sha256 "b75ddc264f0ba5615db7ba217daeb99701ad295353c45f9e95963337ceeeffb2"
+  end
+
```
Comment (p-linnane) [inline Formula/c/codelimit.rb:156]:
```suggestion

```

File: Formula/c/codelimit.rb [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+    sha256 "04ab9d4b9f587c06d801c2abfe9317b77cdf996c65a90d5e84ecc45010823571"
+  end
+
+  resource "zipp" do
+    url "https://files.pythonhosted.org/packages/e2/45/f3b987ad5bf9e08095c1ebe6352238be36f25dd106fde424a160061dce6d/zipp-3.16.2.tar.gz"
+    sha256 "ebc15946aa78bd63458992fc81ec3b6f7b1e92d51c35e6de1c3804e73b799147"
+  end
+
+  def install
+    virtualenv_create(libexec, "python3")
```
Comment (p-linnane) [inline Formula/c/codelimit.rb:173]:
```suggestion

```

File: Formula/c/codelimit.rb [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+  url "https://files.pythonhosted.org/packages/71/db/872656169f3a75e923faf89c4ef57ccf7691722422e6d708f6ced6edbde2/codelimit-0.6.0.tar.gz"
+  sha256 "2791dded3c894a44525bcb9d354088626d1627272137b8273513bfb704efbfea"
+
+  bottle do
+    root_url "https://github.com/getcodelimit/homebrew-codelimit/releases/download/codelimit-0.6.0"
+    sha256 cellar: :any_skip_relocation, ventura:      "c7cfc66afff1c3aa8c944f0bd574563a5c328bf371384fb2f0659d68dce661c4"
+    sha256 cellar: :any_skip_relocation, x86_64_linux: "adf7735830c54a057bd4f0b6db0d9f02b99eb6ec928a9efb86cb8805e0ed4920"
+  end
+
+  depends_on "python3"
```
Comment (p-linnane) [inline Formula/c/codelimit.rb:15]:
```suggestion
  depends_on "python-certifi"
  depends_on "python-typing-extensions"
  depends_on "pygments"
  depends_on "python@3.11"
  depends_on "six"
```

File: Formula/c/codelimit.rb
Comment (p-linnane) [inline Formula/c/codelimit.rb:1]: We'll need to add this to `pypi_formula_mappings.json` as well for the excluded resources.

**Review by p-linnane — CHANGES_REQUESTED** [review]

File: Formula/c/codelimit.rb (hunk 8 above)
Comment (robvanderleek) [reply Formula/c/codelimit.rb]: I've added these dependencies.

File: Formula/c/codelimit.rb
Comment (robvanderleek) [reply Formula/c/codelimit.rb:1]: I've added the excluded resources.

Comment (p-linnane) [reply Formula/c/codelimit.rb:1]: Thanks. Can you please squash all the commits into one with the message `codelimit 0.6.0 (new formula)`?

**Comment by p-linnane** [comment]
Thanks for the PR. Running `brew style --fix codelimit` once more should fix the last style error. Once that's done, can you please squash all the commits into one with the message `codelimit 0.6.0 (new formula)`?

**Comment by robvanderleek** [comment]
> Thanks for the PR. Running `brew style --fix codelimit` once more should fix the last style error. Once that's done, can you please squash all the commits into one with the message `codelimit 0.6.0 (new formula)`?

I've squashed all commits 🙂  Thanks for the support.

**Comment by robvanderleek** [comment]
Hi @p-linnane 

Is there any open review issue holding this PR back? Or will it be merged eventually?

**Review by p-linnane — APPROVED** [review]

**Comment by p-linnane** [comment]
My mistake, I thought this finished running. Let me fix that.

**Comment by p-linnane** [comment]
Also for future reference, branches from organizations cause issues with our merge process. Please open them from normal forks instead.

**Comment by p-linnane** [comment]
@carlocab Can you take a look here? Not sure why the replacement workflow says this isn't mergable.

**Comment by robvanderleek** [comment]
It seems I can't enable "Allow edits from maintainers" for this PR anymore. It was created from a [GH org fork](https://github.com/getcodelimit/homebrew-core). Let me know if I can help, no problem to open a new PR from a user fork if needed.

Thanks for all the support.

**Comment by p-linnane** [comment]
If you can open a new PR from a user fork we can probably get this merged today.

**Comment by robvanderleek** [comment]
Hi @p-linnane 

I've opened a PR from my own (user) fork: https://github.com/Homebrew/homebrew-core/pull/140506

KR, Rob

**Pull request closed without merging**